server/data/data-jpa/src/main/java/org/apache/james/sieve/jpa/model/JPASieveScript.java

```diff
@@ -0,0 +1,188 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.sieve.jpa.model;
+
+import java.time.OffsetDateTime;
+import java.util.Objects;
+import java.util.UUID;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+
+import org.apache.james.sieverepository.api.ScriptContent;
+import org.apache.james.sieverepository.api.ScriptName;
+import org.apache.james.sieverepository.api.ScriptSummary;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+@Entity(name = "JamesSieveScript")
+@Table(name = "JAMES_SIEVE_SCRIPT")
+@NamedQueries({
+        @NamedQuery(name = "findAllByUsername", query = "SELECT sieveScript FROM JamesSieveScript sieveScript WHERE sieveScript.username=:username"),
+        @NamedQuery(name = "findActiveByUsername", query = "SELECT sieveScript FROM JamesSieveScript sieveScript WHERE sieveScript.username=:username AND sieveScript.isActive=true"),
+        @NamedQuery(name = "findSieveScript", query = "SELECT sieveScript FROM JamesSieveScript sieveScript WHERE sieveScript.username=:username AND sieveScript.scriptName=:scriptName")
+})
+public class JPASieveScript {
+
+    public static Builder builder(String username, String scriptName) {
+        return new Builder(username, scriptName);
+    }
+
+    public static ScriptSummary toSummary(JPASieveScript script) {
+        return new ScriptSummary(new ScriptName(script.getScriptName()), script.isActive());
+    }
+
+    public static class Builder {
```

Comment:
The pattern we have chosen for builders is a bit different that this one, here are some remarks matching _James_ builder pattern:
* the constructor should not take parameter
* then, the _Builder_ does not have _final_ attributes
* we are not doing smart things in methods, for example the _activationDateTime_ should be calculated in the _build_ method
* the _build_ method is calling the private constructor of the class by passing all parameters, not _this_

Here is an example you can follow:
https://github.com/linagora/james-project/blob/436e8cde4b35b4662f1a9edf38b7eb2382fb5f54/mailbox/api/src/main/java/org/apache/james/mailbox/model/Attachment.java#L38